sphinx/builders/linkcheck.py

```diff
@@ -706,15 +713,37 @@ def contains_anchor(response: Response, anchor: str) -> bool:
 class AnchorCheckParser(HTMLParser):
     """Specialised HTML parser that looks for a specific anchor."""
 
-    def __init__(self, search_anchor: str) -> None:
+    def __init__(self, search_anchor: str, original_encoded_anchor: str = '') -> None:
+        """Initialize the parser with multiple anchor variations.
+
+        Args:
+            search_anchor: The decoded anchor to search for
+                (e.g., "standard-input/output-stdio")
+            original_encoded_anchor: The original encoded anchor
+                (e.g., "standard-input%2Foutput-stdio")
+        """
         super().__init__()
 
-        self.search_anchor = search_anchor
+        # Create variations of the anchor to check
+        self.search_variations = {
+            search_anchor,  # decoded (current behavior)
+        }
```

Comment:
I'd recommend that we try to search the anchor name variations in the order listed in the WHATWG HTML spec for `Scrolling to a fragment`: https://html.spec.whatwg.org/commit-snapshots/4467ddf3235732614a0202729d6ec0e04c33b597/#scrolling-to-a-fragment

To do that, I think we may need to use a different datastructure than Python's `set`, because it doesn't necessarily iterate over elements in the order they were added/inserted.

Reply:
My mistake; we only use the `search_variations` set attribute for presence-checking as we progress our way through each HTML document.  Iteration order does not matter.  So: I think `set` is fine, and we do not need to worry about the order that we add items -- provided that the resulting collection is equivalent to the anchor names a browser could check for.